ocs_ci/utility/openshift_console.py

```diff
@@ -0,0 +1,112 @@
+import os
+import logging
+from datetime import datetime
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import (
+    CommandFailed,
+    OpenshiftConsoleSuiteNotDefined,
+    UnsupportedBrowser,
+)
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility.utils import get_kubeadmin_password, run_cmd
+
+logger = logging.getLogger(__name__)
+
+
+class OpenshiftConsole:
+
+    def __init__(self, browser=constants.CHROME_BROWSER):
+        self.browser = browser
+        self.console_path = config.RUN['openshift_console_path']
+        self.env_vars = {}
+        self.setup_console_prereq()
+
+    def setup_console_prereq(self):
+        """
+        Setup openshift console prerequisites
+
+        Raises:
+            UnsupportedBrowser: in case the browser is not supported
+        """
+        # TODO: add support for other browsers
+        if self.browser not in constants.SUPPORTED_BROWSERS:
+            raise UnsupportedBrowser(
+                f"Support for {self.browser} hasn't been implemented yet!"
+            )
+        if self.browser == constants.CHROME_BROWSER:
+            chrome_branch_base = config.RUN.get("force_chrome_branch_base")
+            chrome_branch_sha = config.RUN.get("force_chrome_branch_sha256sum")
+            self.env_vars["FORCE_CHROME_BRANCH_BASE"] = chrome_branch_base
+            self.env_vars["FORCE_CHROME_BRANCH_SHA256SUM"] = chrome_branch_sha
+
+        htpasswd_secret = OCP(
+            kind="Secret", resource_name=constants.HTPASSWD_SECRET_NAME
+        )
+        try:
+            htpasswd_secret.get()
+            logger.info("Htpasswd secret is already set! Skipping secret setup")
+        except CommandFailed:
```

Comment:
maybe would be good to examine the exception and check if the message is about secret not set and in case not, re-raise

Reply:
It's about calling command oc get secret secret-name . So this command fill fail 99% only if the object doesn't exist and this case I would like to do the rest of preparation. In case some other issue with API or whatever else we will see error in other call. So I think that not necessary to add another logic here.